com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceGetter.java

```diff
@@ -37,24 +36,19 @@ public ServiceGetter(@Qualifier("serviceInstanceGetter") AbstractServiceGetter s
         return serviceInstance;
     }
 
-    public List<ServiceInstanceWithServiceKeys> getServiceInstancesWithServiceKeys(CloudControllerClient client,
-                                                                                   List<CloudServiceInstance> services) {
-        return services.stream()
-                       .map(service -> buildServiceInstanceWithServiceKeys(service, client))
-                       .collect(Collectors.toList());
+    public ServiceInstanceWithServiceKeys getServiceInstancesWithServiceKeys(CloudControllerClient client, CloudServiceInstance service) {
+        return buildServiceInstanceWithServiceKeys(service, client);
```

Comment:
If buildServiceInstanceWithServiceKeys(service, client) isn't called elsewhere you can just move the code to getServiceInstancesWithServiceKeys.